### PR TITLE
TeamCity v6.0.0 feature branch testing

### PR DESCRIPTION
**Pull request by SarahFrench**
This PR enables nightly testing of the v6.0.0 feature branch

## Schedule:

- On Saturday-Wednesday nightly tests will run off main branch of `google` and `google-beta` repos as ususal.
- On Thursdays the GA tests will run off the feature branch. Beta tests are off main as usual.
- On Fridays the Beta tests will run off the feature branch. GA tests are off main as usual.

## What it looks like:

- There's a new project put inside the root "Google Cloud" project called "6.0.0 Major Release Testing"
- Inside there is a Google and Google Beta project, which each contain Nightly Test projects
- Those Nightly Test projects contain builds for all packages

Here is an example deployment: https://hashicorp.teamcity.com/project/TerraformProviders_SarahSTestProjectTpg?projectTab=overview&mode=builds

## Screenshots of project layout :

Screenshot 2024-07-03 at 16 47 50

## Screenshots of the Beta cron triggers:

1. main branch

Screenshot 2024-07-03 at 16 54 55

2. FEATURE-BRANCH-major-release-6.0.0 branch

Screenshot 2024-07-03 at 16 54 24

**Release Note Template for Downstream PRs (will be copied)**

```release-note:none

```

### CODE REVIEW

File: mmv1/third_party/terraform/.teamcity/components/projects/reused/mm_upstream.kt
Comment (shuyama1) [inline mmv1/third_party/terraform/.teamcity/components/projects/reused/mm_upstream.kt:28]: Just want to confirm if I understand it correctly - Is the modification to the MM upstream config for stopping VCR sweepers on release branch test days?

File: mmv1/third_party/terraform/.teamcity/components/builds/build_triggers.kt
Comment (shuyama1) [inline mmv1/third_party/terraform/.teamcity/components/builds/build_triggers.kt:18]: I assume that all the changes in this PR need to be synced to the 6.0.0 release branch first before it can be used for release branch testing?

**Review by shuyama1 — COMMENTED** [review]
LGTM overall. I only have some small questions. Thank you so much for the work! 

Also would you mind granting me access to https://hashicorp.teamcity.com/project/TerraformProviders_SarahSTestProjectTpg?projectTab=overview&mode=builds?

File: mmv1/third_party/terraform/.teamcity/components/projects/reused/mm_upstream.kt
Comment (SarahFrench) [reply mmv1/third_party/terraform/.teamcity/components/projects/reused/mm_upstream.kt:28]: I think I just wanted to make the functions used to create nightly testing and upstream testing Projects in TeamCity consistent, in that they both require the cron trigger to explicitly passed in.

Currently the sweepers for the VCR project will run independently of what's happening in Nightly Test projects.

Am I right that the v6.0.0 testing should use the nightly test projects and not the VCR project in GCP?

File: mmv1/third_party/terraform/.teamcity/components/builds/build_triggers.kt
Comment (SarahFrench) [reply mmv1/third_party/terraform/.teamcity/components/builds/build_triggers.kt:18]: I don't think so - this PR would be merged into main and TeamCity will use the configuration present on main branch of the TPG repo to control what's in TeamCity.

I believe the only coordination required for these types of changes is that the [feature branch(es) being referenced](https://github.com/GoogleCloudPlatform/magic-modules/blob/ef55f56f8759510005768fda6bf22a04d44b1b9f/mmv1/third_party/terraform/.teamcity/components/projects/feature_branches/FEATURE-BRANCH-major-release-6.0.0.kt#L18) would need to exist before we merge changes to the config.

**Comment by SarahFrench** [comment]
@shuyama1 Thanks for the review- I've given you access to the project!

**Comment by SarahFrench** [comment]
Actually I've found an issue with the config - something is wrong with how the v6.0 testing project accesses GitHub branches 🤦  - marking as draft while I look into it

**Comment by SarahFrench** [comment]
I think that's fixed - you should have access to https://hashicorp.teamcity.com/project/TerraformProviders_SarahSTestProject?projectTab=overview&mode=builds

**Comment by SarahFrench** [comment]
I'll save merging this until Wednesday, to avoid any potential disruption to cutting a new release

File: mmv1/third_party/terraform/.teamcity/components/projects/reused/mm_upstream.kt
Comment (shuyama1) [reply mmv1/third_party/terraform/.teamcity/components/projects/reused/mm_upstream.kt:28]: Oh, I see, it's for consistency. 

> Am I right that the v6.0.0 testing should use the nightly test projects and not the VCR project in GCP?

Yes, correct! The v6.0.0 testing should use the nightly test projects, same as nightly runs

File: mmv1/third_party/terraform/.teamcity/components/builds/build_triggers.kt
Comment (shuyama1) [reply mmv1/third_party/terraform/.teamcity/components/builds/build_triggers.kt:18]: Nice! Thanks for the explanation. I must be confused with some testing step we did in the past that we needed to have the configuration checked into the branches.

**Review by shuyama1 — APPROVED** [review]
LGTM! The TeamCity testing setup looks good as well. Thank you so much!

**Comment by SarahFrench** [comment]
Thanks! I'll follow this up with a separate PR that documents how we can add feature branch testing for v7.0+

And I'll merge this PR tomorrow once the nightlies complete

**Pull request merged**
